core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -456,4 +473,61 @@ public void close() {
       // Nothing to clean up
     }
   }
+
+  /**
+   * Periodically check if we need to take daily master metadata snapshot.
+   */
+  private final class DailyBackupHeartbeatExecutor implements HeartbeatExecutor {
```

Comment:
We want to run this at the same time each day, so `ScheduledExecutorService` might work better. Otherwise we don't know what time within the 1-hour span it will run, and if we're unlucky we might check right before and right after the hour, or twice in the same hour.